### PR TITLE
Statistics

### PR DESCRIPTION
**Pull request by lf94**

### CODE REVIEW

**Comment by lf94** [comment]
Blocked by https://github.com/alephium/explorer-backend/issues/169

File: src/components/StatisticBlock.tsx
Comment (mvaivre) [inline src/components/StatisticBlock.tsx:35]: So, some "conventions" we've agreed on with @nop33 (conventions, yay! :D) :
- We always put the styled components after the main exported react component (which is `StatisticBlock` here)
- We put the types / interface first.

So we always have 1) Prop types 2) exported component 3) styling.

File: src/components/StatisticBlock.tsx
Comment (mvaivre) [inline src/components/StatisticBlock.tsx:57]: As we don't use `children`, we don't need to use `FC`. We can just go like: 

```ts
const StatisticBlock = ({ title, primary, secondary, isLoading }: Props) => 
```

And as this is a "dump component" (presentational only, no logic), we don't need the brackets and the return statement :)

File: src/components/StatisticBlock.tsx
Comment (lf94) [reply src/components/StatisticBlock.tsx:35]: o7 roger that

File: src/components/StatisticBlock.tsx
Comment (lf94) [reply src/components/StatisticBlock.tsx:57]: Habit to just make everything FC<>, even if a dumb component, will remove :+1:

**Comment by lf94** [comment]
To test the code you'll have to check out and run https://github.com/alephium/explorer-backend/tree/feature/nb-of-tx-endpoint, and generate your own alephium-js package to make use of the new OpenAPI spec.

File: src/components/StatisticBlock.tsx
Comment (lf94) [reply src/components/StatisticBlock.tsx]: Gah, I still need to do this.

I would still use `FC` but you're right about removing the `return` - it's 2 characters for a more accurate type. If you feel strongly about omitting it I'll do it though, I just don't see any good reason other than "it adds more semantics to the code", which is a good reason I guess but not one I agree with :slightly_smiling_face:

Comment (nop33) [reply src/components/StatisticBlock.tsx]: I haven't seen any benefit of having a `FC` over a normal function so far. So we decided to drop it when not needed and save one line of imports.

File: README.md
Comment (nop33) [inline README.md:7]: I think an `npm run format` would update this file.

File: package.json
Comment (nop33) [inline package.json:10]: This should be changed before merging. Also, using `npm link` removes the need to change this.

File: src/components/StatisticBlock.tsx
Comment (nop33) [inline src/components/StatisticBlock.tsx:48]: Nit: I would personally do it this way:
1. Remove `FC` since we don't need `children` (same for other components in other files)
2. Remove `return` keyword since it's a dumb component
3. Use a ternary operator to avoid using a fragment

```suggestion
const StatisticBlock = ({ title, primary, secondary, isLoading }: Props) =>
  isLoading ? (
    <BlockCenteredContent>
      <LoadingSpinner />
    </BlockCenteredContent>
  ) : (
    <Block>
      <Title>{title}</Title>
      <Primary>{primary}</Primary>
      <Secondary>{secondary}</Secondary>
    </Block>
  )
```

File: src/components/Statistics.tsx
Comment (nop33) [inline src/components/Statistics.tsx:89]: This should work as well, right? (also below)

```suggestion
        title="Hashrate"
```

File: src/components/StatisticBlock.tsx
Comment (lf94) [reply src/components/StatisticBlock.tsx]: Fair enough! I'll omit it.

----

The comparison I see is you wouldn't just leave a function untyped normally, because it's bad type safety: `const f = (s: string): string => 'takes and returns a string';`
So I don't (personally) compromise when having a function which returns an element either: `const F = (p: Props):ReactElement => <></>;` - `FC` is just shorthand for this. You never see this as an issue though because TypeScript is inferring that `<></>` is an `Element` so it deduces that `F` returns an `Element`, but still, it's not good because what if you made a mistake in the function which returns a `string` or a type other than `Element`?

File: src/components/Statistics.tsx
Comment (nop33) [inline src/components/Statistics.tsx:122]: Are these `div`s really needed? Their parent seems to be just another `div`. Would it not work well if it was just `secondary="Total"`? (also below)

File: src/sections/TransactionInfoSection.tsx
Comment (nop33) [inline src/sections/TransactionInfoSection.tsx:52]: Isn't there a similar interface in alephium-js?

Comment (lf94) [reply src/sections/TransactionInfoSection.tsx:52]: Yep, very similar, but the differences were causing problems. Feel free to suggest code changes using `TransactionLike`, `Transaction`, `UnconfirmedTransaction`, `ConfirmedTransaction`, but I've tried and it isn't trivial.

Comment (nop33) [reply src/sections/TransactionInfoSection.tsx:52]: Isn't this exactly the same? https://github.com/alephium/alephium-js/blob/upgrade-schemas/api/api-explorer.ts#L132-L144

Comment (lf94) [reply src/sections/TransactionInfoSection.tsx:52]: Maybe I had a brainfart moment because yep, my visual cortex is interpreting this as exactly the same. :rofl:

File: package.json
Comment (lf94) [reply package.json:10]: It will need to be updated no matter what because it relies on api-explorer.ts updates - I'll leave this comment open until that happens

**Comment by lf94** [comment]
I've addressed the suggestions - will ping everyone when the backend updates are ready! Thank you for the great comments!

File: README.md
Comment (nop33) [inline README.md:35]:
```suggestion
This project uses [@alephium/sdk](https://github.com/alephium/js-sdk).
```

File: README.md
Comment (nop33) [inline README.md:36]:
```suggestion
It's suggested to download it to refer to it as documentation on
```

**Review by nop33 — CHANGES_REQUESTED** [review]

**Pull request merged**
